python/ray/tempfile_services.py

```diff
@@ -0,0 +1,288 @@
+    you get around to creating it, someone else may have beaten you to
+    the punch.
+    """
+    sockets_dir = get_sockets_dir_path()
+
+    if suffix is None:
+        raylet_socket_name = make_inc_temp(
+            prefix='scheduler_', directory_name=sockets_dir)
+    else:
+        raylet_socket_name = os.path.join(sockets_dir,
+                                          'scheduler_{}'.format(suffix))
+    return raylet_socket_name
+
+
+def get_random_ipython_notebook_path(port):
+    """Get a new random ipython notebook path"""
+
+    notebook_filepath = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "WebUI.ipynb")
+    # We copy the notebook file so that the original doesn't get modified by
+    # the user.
+    notebook_name = make_inc_temp(
+        suffix='.ipynb', prefix='ray_ui_', directory_name=get_temp_root())
+    new_notebook_filepath = os.path.join(get_logs_dir_path(), notebook_name)
+    shutil.copy(notebook_filepath, new_notebook_filepath)
+    new_notebook_directory = os.path.dirname(new_notebook_filepath)
+    token = ray.utils.decode(binascii.hexlify(os.urandom(24)))
+    webui_url = ("http://localhost:{}/notebooks/{}?token={}".format(
+        port, notebook_name, token))
+    return new_notebook_directory, webui_url, token
+
+
+def get_random_temp_redis_config_path():
+    """Get a temp name of the redis config file."""
+    redis_config_name = make_inc_temp(
+        prefix='redis_conf_', directory_name=get_temp_root())
+    return redis_config_name
+
+
+def new_log_files(name, redirect_output):
+    """Generate partially randomized filenames for log files.
+
+    Args:
+        name (str): descriptive string for this log file.
+        redirect_output (bool): True if files should be generated for logging
+            stdout and stderr and false if stdout and stderr should not be
+            redirected.
+
+    Returns:
+        If redirect_output is true, this will return a tuple of two
+            filehandles. The first is for redirecting stdout and the second is
+            for redirecting stderr. If redirect_output is false, this will
+            return a tuple of two None objects.
+    """
+    if not redirect_output:
+        return None, None
+
+    # Create a directory to be used for process log files.
+    logs_dir = get_logs_dir_path()
+    # Create another directory that will be used by some of the RL algorithms.
+    # TODO: Move it into where we need such a directory.
+    get_rl_dir_path()
+
+    date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S")
+    prefix = '{}-{}-'.format(name, date_str)
+
+    log_stdout = make_inc_temp(
+        suffix='.out', prefix=prefix, directory_name=logs_dir)
+    log_stderr = make_inc_temp(
+        suffix='.err', prefix=prefix, directory_name=logs_dir)
+    # Line-buffer the output (mode 1)
+    log_stdout_file = open(log_stdout, "a", buffering=1)
+    log_stderr_file = open(log_stderr, "a", buffering=1)
```

Comment:
Is it safe to not close these files? 
I used to see a lot of error messages in tests like this (python 3 only):
```
/home/travis/.local/lib/python3.6/site-packages/ray-0.4.0-py3.6-linux-x86_64.egg/ray/services.py:1510: ResourceWarning: unclosed file <_io.TextIOWrapper name='/tmp/raylogs/worker_0_0-2018-07-06_08-19-29-04610.out' mode='a' encoding='UTF-8'> "worker_{}_{}".format(i, j), redirect_output)
```
But these files are created for other processes, I'm not sure what is the best way to close them.

Reply:
These messages have been seen in previous tests. This may be because some process just created some log files and get killed in the test.